src/test/ui/typeck/issue-84768.stderr

```diff
@@ -12,6 +12,10 @@ LL |     <F as FnOnce(&mut u8)>::call_once(f, 1)
    |
    = note: expected tuple `(&mut u8,)`
                found type `{integer}`
+help: use a trailing comma to create a tuple with one element
+   |
+LL |     <F as FnOnce(&mut u8)>::call_once(f, (1,))
+   |                                          + ++
```

Comment:
This suggestion seems to be incorrect, though this bug existed before your change. Upon re-reading the code that determines whether the suggestion should be emitted, I noticed that it does not actually check that the tuple's element is the same as the expected type 😅 

I'm okay approving this as-is since this bug was pre-existing, but do you want to try fixing it? Otherwise, I can open a followup PR to try to do it.

Reply:
Ah yeah I see - I'd like to have a go at fixing it. I'll do it as a separate commit, but is it otherwise ok to be included in this PR?

Reply:
Yeah, that seems fine :)

Reply:
This is [fixed now](https://github.com/rust-lang/rust/pull/91530/commits/91a43f04237dc9fe839fe8466d459e4f215d02fc?diff=split&w=1) - some nested if / if-lets, but I kept the code local like the other match arms (happy to pull out to a function though)